pkg/core/extension_store.go

```diff
@@ -493,6 +516,8 @@ func (store *ExtensionStore) UpdateExtension(ctx context.Context, extension *dom
 	if err != nil {
 		return err
 	}
+	extension.Registered = extRecord.Registered
```

Comment:
Why `Registered`?

Reply:
This is to avoid overwriting the registered timestamp when the extension is updated

Reply:
I see. The `Registered` field is generated and we can't expect user to provide it on update.

But this brings another question - what if user wants to update just one entry of the object (say, Description), and provides only this in the POST method. With current implementation, all other fields will be deleted, right? Should this be the intended behavior of the  `Update` methods?

Reply:
Good question.

To implement this type of surgical update, we need to make all the fields in the design objects optional (already are) and also to not give them a default value. This will force goa to generate pointers as attributes, instead of actual values as they are now.

The other solution is to push this mechanism further out, into the CLI and UI (when available).

Changing everything to pointers now is going to be a PITA to refactor, but I'll give it a shot.

Reply:
Done. Was easier than I thought.